src/Vault.sol

```diff
@@ -1140,10 +1113,18 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
    * @return uint256 Amount of underlying assets that can deposited
    */
   function _maxDeposit(uint256 _depositedAssets) internal view returns (uint256) {
+    uint256 _depositedAssets = _totalSupply();
+    uint256 _withdrawableAssets = _totalAssets();
     uint256 _vaultMaxDeposit = UINT112_MAX - _depositedAssets;
     uint256 _yieldVaultMaxDeposit = _yieldVault.maxDeposit(address(this));
 
-    return _yieldVaultMaxDeposit < _vaultMaxDeposit ? _yieldVaultMaxDeposit : _vaultMaxDeposit;
+    // Vault shares a minted 1:1 when the vault is collateralized,
+    // so maxDeposit and maxMint return the same value
+    if (_isVaultCollateralized(_depositedAssets, _withdrawableAssets)) {
+      return _yieldVaultMaxDeposit < _vaultMaxDeposit ? _yieldVaultMaxDeposit : _vaultMaxDeposit;
+    }
```

Comment:
With this addition, we no longer need to check for collateralization on calls that revert when over `maxDeposit`.

ex (in `_depositAssets` function):
![image](https://github.com/GenerationSoftware/pt-v5-vault/assets/40277611/01156612-946d-41f1-8d9f-9aa01f817851)